### PR TITLE
archive tool

### LINKED ISSUE #1263: Tool to archive content
Suppose you're messing around in the [content](https://github.com/mdn/content) and you realize "This page is as ancient as the Babylonians roamed the earth". The right thing is to do...:

1. `mkdir -p /path/to/mdn/archived-content/files/en-us/path/to/ancient/page/`
1. `cp files/en-us/path/to/ancient/page/index.html /path/to/mdn/archived-content/files/en-us/path/to/ancient/page/`
1. `git rm files/en-us/path/to/ancient/page/index.html`
1. `cd /path/to/mdn/archived-content`
1. `git add files/en-us/path/to/ancient/page/`

I don't think we should write a tool that attempts to do too much `git` commands, but we should at least have a tool that can do what's necessary with the files, and then the human will do the necessary `git` commands manually.

### LINKED ISSUE #3639: archive tool
We need a way to copy content into mdn/archived-content before deleting it from mdn/content (or mdn/translated-content).

Also, this CLI tool should make it easy to create a redirect. Perhaps to the GitHub file browser [example](https://github.com/mdn/archived-content/tree/b7b6516b2b522533e8ba51cb944700ace7dace4d/files/en-us/mozilla/tech/xpcom/fun_with_xbl_and_xpconnect).

### PR DESCRIPTION
**Pull request by peterbe**
Fixes #3639
Fixes #1263

Leaving it as a draft for now because I don't want to review and merge it until I've tested it out properly on real content that needs to be moved from mdn/content (or mdn/translated-content) to mdn/archived-content.

### CODE REVIEW

**Comment by peterbe** [comment]
I'm not sure what I want to do with this. It works. It's not entirely clear if we'll re-use this in the long-tail future when you want to "remove" some documents.

**Comment by peterbe** [comment]
@escattone I used this to create all the migrations for the documents listed here: https://docs.google.com/document/d/1exNMAfz58uOifkKgJp3itrWBLwvOX2kymhpMDVoqpE8/edit

File: content/document.js [hunk 1]
```diff
@@ -193,6 +186,24 @@ function archive(
   }
 
   saveFile(getHTMLPath(folderPath), trimLineEndings(renderedHTML), metadata);
+
+  // Next we need to every single other file that isn't index.html or index.md
```
Comment (escattone) [inline content/document.js:190]: Did you mean this?
```suggestion
  // Next we need to copy every single file that isn't index.html or index.md
```

**Review by escattone — APPROVED** [review]
Nice, thanks for this @peterbe! Just the one comment nit.

**Pull request merged**